subt/ros/proxy/traversability.cc

```diff
@@ -0,0 +1,769 @@
+      // the robot can drive over. These dips occur, for example, when crossing
+      // a rail submerged into the surface.
+      float max_dip_down;
+      // A dipped obstacle, such as a rail, can extrude slightly above the
+      // surrounding surface.
+      float max_dip_up;
+      // We insert a synthetic obstacle this far into the map when there is
+      // no traversable map ahead. For the system to work, a horizontally
+      // placed robot needs to have this obstacle at the bottom of depth
+      // camera vision just below the ground surface.
+      float synthetic_obstacle_distance;
+      // To feel comforrtable, we need to see ground somewhere between these
+      // two distance thresholds.
+      float visible_ground_min, visible_ground_max;
+      // Obstacles more than this above ground are ignored, because they are not
+      // relevant. the robot can go under them,
+      float robot_height;
+    } config_;
+
+    // Maps are keyed by frame_ids of individual sensors.
+    std::map<std::string, int> observation_seq_id_;
+    std::map<std::string, std::vector<Observation>> observations_;
+
+    ros::Timer publish_timer_;
+    ros::Publisher points_publisher_;
+    ros::Publisher scan_publisher_;
+
+    std::optional<tf::StampedTransform> GetTransform(
+        const std::string& target_frame, const std::string& source_frame, const ros::Time& when) const;
+    void InsertObservation(
+        Observation&& observation, const std::string& frame_id,
+        size_t max_num_observations_per_source);
+    void HandleDepth(
+        const cv::Mat& img, const std::string& camera_frame_id,
+        const ros::Time& stamp);
+
+    void OnCameraInfo(const sensor_msgs::CameraInfo::ConstPtr& msg);
+    void OnDepth(const sensor_msgs::Image::ConstPtr& msg);
+    void OnCompressedDepth(const rtabmap_ros::RGBDImage::ConstPtr& msg);
+    void OnScan(const sensor_msgs::LaserScan::ConstPtr& msg);
+    void OnTimer(const ros::TimerEvent& event);
+};
+
+bool Traversability::Init()
+{
+  ros_handle_.param<std::string>("robot_frame_id", config_.robot_frame_id, "");
+  ros_handle_.param<std::string>("world_frame_id", config_.world_frame_id, "odom");
+  ros_handle_.param("max_depth_observations", config_.max_depth_observations, 30 * 6);
+  ros_handle_.param("max_depth", config_.max_depth, 10.0f);
+  ros_handle_.param("depth_image_stride", config_.depth_image_stride, 2);
+  ros_handle_.param("depth_subsampling", config_.depth_subsampling, 123);
+  ros_handle_.param("horizontal_pixel_offset", config_.horizontal_pixel_offset, 12);
+  ros_handle_.param("vertical_pixel_offset", config_.vertical_pixel_offset, 12);
+  ros_handle_.param("max_neighbor_distance", config_.max_neighbor_distance, 0.3f);
+  ros_handle_.param("max_relative_z", config_.max_relative_z, 40.f);
+  ros_handle_.param("max_scan_observations", config_.max_scan_observations, 20 * 6);
+  ros_handle_.param("max_lidar_range", config_.max_lidar_range, 0.45f);
```

Comment:
Hm, this should be something higher, like 10 m.

Reply:
now I am not sure if it is the real lidar range or range where we want to prefer lidar to RGBD? Also note K2 has config 0.5m in ROS launch file